### PR TITLE
Role enum (and enum updates generally)

### PR DESCRIPTION
**Pull request by matoszz**
This may be slightly preliminary, but we have various needs to set a general role on the `User` object which would simplify many operations if we were able to set it on the Viewer context - FGA may be the better check for this exercise but need to explore how best to represent fine grained permissions on the user's access context

### CODE REVIEW

File: internal/ent/enums/auth_provider.go [hunk 1]
```diff
@@ -3,25 +3,28 @@ package enums
...
+	AuthProviderWebauthn AuthProvider = "WEBAUTHN"
+	// AuthProviderInvalid is the default value for the AuthProvider enum
+	AuthProviderInvalid AuthProvider = "INVALID"
 )
 
 // Values returns a slice of strings that represents all the possible values of the AuthProvider enum.
 // Possible default values are "CREDENTIALS", "GOOGLE", "GITHUB", and "WEBAUTHN"
 func (AuthProvider) Values() (kinds []string) {
-	for _, s := range []AuthProvider{Credentials, Google, GitHub, Webauthn} {
+	for _, s := range []AuthProvider{AuthProviderCredentials, AuthProviderGoogle, AuthProviderGitHub, AuthProviderWebauthn, AuthProviderInvalid} {
```
Comment (golanglemonade) [inline internal/ent/enums/auth_provider.go:27]: `AuthProviderInvalid` should not be included as a valid auth provider for the schema 

```suggestion
	for _, s := range []AuthProvider{AuthProviderCredentials, AuthProviderGoogle, AuthProviderGitHub, AuthProviderWebauthn} {
```

File: internal/ent/enums/documenttype.go [hunk 2]
```diff
@@ -13,12 +13,14 @@ var (
...
 	Document DocumentType = "DOCUMENT"
+	// DocumentTypeInvalid is the default value for the DocumentType enum
+	DocumentTypeInvalid DocumentType = "INVALID"
 )
 
 // Values returns a slice of strings that represents all the possible values of the DocumentType enum.
 // Possible default values are "ROOTTEMPLATE", "DOCUMENT"
 func (DocumentType) Values() (kinds []string) {
-	for _, s := range []DocumentType{RootTemplate, Document} {
+	for _, s := range []DocumentType{RootTemplate, Document, DocumentTypeInvalid} {
```
Comment (golanglemonade) [inline internal/ent/enums/documenttype.go:23]: Same here, these are the valid values in the database, invalid should not be included 

```suggestion
	for _, s := range []DocumentType{RootTemplate, Document} {
```

File: internal/ent/enums/invitestatus.go [hunk 3]
```diff
@@ -19,7 +19,7 @@ var (
 // Values returns a slice of strings that represents all the possible values of the InviteStatus enum.
 // Possible default values are "INVITATION_SENT", "APPROVAL_REQUIRED", "INVITATION_ACCEPTED", and "INVITATION_EXPIRED"
 func (InviteStatus) Values() (kinds []string) {
-	for _, s := range []InviteStatus{InvitationSent, ApprovalRequired, InvitationAccepted, InvitationExpired} {
+	for _, s := range []InviteStatus{InvitationSent, ApprovalRequired, InvitationAccepted, InvitationExpired, InviteInvalid} {
```
Comment (golanglemonade) [inline internal/ent/enums/invitestatus.go:22]:
```suggestion
	for _, s := range []InviteStatus{InvitationSent, ApprovalRequired, InvitationAccepted, InvitationExpired} {
```

File: internal/ent/enums/join_policy.go [hunk 4]
```diff
@@ -9,16 +9,22 @@ import (
...
+	JoinPolicyInviteOrApplication JoinPolicy = "INVITE_OR_APPLICATION"
+	// JoinPolicyInvalid is the default value for the JoinPolicy enum
+	JoinPolicyInvalid JoinPolicy = "INVALID"
 )
 
 // Values returns a slice of strings that represents all the possible values of the JoinPolicy enum.
 // Possible default values are "OPEN", "INVITE_ONLY", "APPLICATION_ONLY", and "INVITE_OR_APPLICATION".
 func (JoinPolicy) Values() (kinds []string) {
-	for _, s := range []JoinPolicy{Open, InviteOnly, ApplicationOnly, InviteOrApplication} {
+	for _, s := range []JoinPolicy{JoinPolicyOpen, JoinPolicyInviteOnly, JoinPolicyApplicationOnly, JoinPolicyInviteOrApplication, JoinPolicyInvalid} {
```
Comment (golanglemonade) [inline internal/ent/enums/join_policy.go:27]:
```suggestion
	for _, s := range []JoinPolicy{JoinPolicyOpen, JoinPolicyInviteOnly, JoinPolicyApplicationOnly, JoinPolicyInviteOrApplication} {
```

File: internal/ent/enums/region.go [hunk 5]
```diff
@@ -18,7 +18,7 @@ var (
 // Values returns a slice of strings that represents all the possible values of the Region enum.
 // Possible default values are "AMER", "EMEA", and "APAC"
 func (Region) Values() (kinds []string) {
-	for _, s := range []Region{Amer, Emea, Apac} {
+	for _, s := range []Region{Amer, Emea, Apac, InvalidRegion} {
```
Comment (golanglemonade) [inline internal/ent/enums/region.go:21]:
```suggestion
	for _, s := range []Region{Amer, Emea, Apac} {
```

File: internal/ent/enums/status.go [hunk 6]
```diff
@@ -9,17 +9,17 @@ import (
...
+	UserStatusDeactivated UserStatus = "DEACTIVATED"
+	UserStatusSuspended   UserStatus = "SUSPENDED"
+	UserStatusInvalid     UserStatus = "INVALID"
 )
 
 // Values returns a slice of strings that represents all the possible values of the UserStatus enum.
 // Possible default values are "ACTIVE", "INACTIVE", "DEACTIVATED", and "SUSPENDED".
 func (UserStatus) Values() (kinds []string) {
-	for _, s := range []UserStatus{Active, Inactive, Deactivated, Suspended} {
+	for _, s := range []UserStatus{UserStatusActive, UserStatusInactive, UserStatusDeactivated, UserStatusSuspended, UserStatusInvalid} {
```
Comment (golanglemonade) [inline internal/ent/enums/status.go:22]:
```suggestion
	for _, s := range []UserStatus{UserStatusActive, UserStatusInactive, UserStatusDeactivated, UserStatusSuspended} {
```

File: internal/ent/enums/role.go [hunk 7]
```diff
@@ -9,16 +9,17 @@ import (
...
 
 var (
-	RoleOwner  Role = "OWNER"
-	RoleAdmin  Role = "ADMIN"
-	RoleMember Role = "MEMBER"
-	Invalid    Role = "INVALID"
+	RoleOwner   Role = "OWNER"
+	RoleAdmin   Role = "ADMIN"
+	RoleMember  Role = "MEMBER"
+	RoleUser    Role = "USER"
```
Comment (golanglemonade) [inline internal/ent/enums/role.go:15]: What is the difference between `user` and `member` here?

File: internal/ent/enums/role.go [hunk 8]
```diff
@@ -9,16 +9,17 @@ import (
...
+	RoleInvalid Role = "INVALID"
 )
 
 // Values returns a slice of strings that represents all the possible values of the Role enum.
-// Possible default values are "ADMIN", and "MEMBER".
+// Possible default values are "ADMIN", "MEMBER", "OWNER"
 func (Role) Values() (kinds []string) {
-	for _, s := range []Role{RoleAdmin, RoleMember} {
+	for _, s := range []Role{RoleAdmin, RoleMember, RoleOwner, RoleUser} {
```
Comment (golanglemonade) [inline internal/ent/enums/role.go:22]: Owner is not always a valid role, its added to the values for organizations, but we don't have that as a valid role in groups which is why its not set in the default here. 

```suggestion
	for _, s := range []Role{RoleAdmin, RoleMember, RoleUser} {
```

File: internal/ent/enums/role.go (hunk 7 above)
Comment (matoszz) [reply internal/ent/enums/role.go:15]: A user in the system who is not a "member" of something

File: internal/graphapi/models_test.go [hunk 9]
```diff
@@ -436,11 +436,6 @@ func (gm *GroupMemberBuilder) MustNew(ctx context.Context, t *testing.T) *ent.Gr
 		gm.UserID = user.ID
 	}
 
-	role := enums.ToRole(gm.Role)
```
Comment (golanglemonade) [inline internal/graphapi/models_test.go:439]: Why remove the role setting from here? (and line 452?)

**Review by golanglemonade — APPROVED** [review]

**Pull request merged**